metastore/src/main/scala/com/gu/tableversions/metastore/VersionPaths.scala

```diff
@@ -0,0 +1,35 @@
+package com.gu.tableversions.metastore
+
+import java.net.URI
+
+import com.gu.tableversions.core.{Partition, PartitionVersion, VersionNumber}
+
+/**
+  * Encodes the mapping between version numbers and storage paths.
+  */
+object VersionPaths {
+
+  /**
+    * @return the fully resolved path for collection of versioned path
+    */
+  def resolveVersionedPartitionPaths(
+      partitionVersions: List[PartitionVersion],
+      tableLocation: URI): Map[Partition, URI] = {
+
+    partitionVersions.map { partitionVersion =>
```

Comment:
I know this is just a reference implementation at this point, but (nitpick): would it be tidier to have
```scala
def resolveVersionedPartitionPath(partitionVersion: PartitionVersion, tableLocation: URI): URI = ???
```

which is then called by `resolveVersionedPartitionPaths`?

Reply:
I guess that's what `pathFor` is - I was going to make that private, but maybe I should rename it to `resolveVersionedPartitionPath` instead?

Reply:
Yeah, I don't feel strongly either way, just raising it as something to consider. Looks good as it is 👍